### PR TITLE
Add document migration flow (WIP)

### PR DESCRIPTION
**Pull request by ronaldtse**
@Intelligent2013 can you please help integrate the stepmod2mn migration processes for all the content treated by this repository?

This includes:
* convert all XML files into adoc files for the path `data/resource_docs/*/resource.xml`
* generate SVG images in the tree data
* generate SVG image for Express Imagemap XML and image

The location to add code is at:
https://github.com/metanorma/iso-10303-stepmod-annotate/blob/8419b60b654f34e868697cf99f30bb13e7f49110/.github/workflows/annotate.yml#L64-L80

Thanks!

### CODE REVIEW

File: .github/workflows/annotate.yml [hunk 1]
```diff
@@ -61,6 +61,24 @@ jobs:
         run: |
           stepmod-annotate-all .
 
+      - name: Download stepmod2mn and convert SVGs
+        working-directory: iso-10303-stepmod-wg12
+        run: |
+          # TODO
+          wget https://github.com/metanorma/stepmod2mn/releases/download/v1.8/stepmod2mn-1.8.jar
```
Comment (ronaldtse) [inline .github/workflows/annotate.yml]: Should we use 1.8 or 2.2?

Comment (Intelligent2013) [reply .github/workflows/annotate.yml]: The latest version is 1.8 (https://github.com/metanorma/stepmod2mn/releases)

**Comment by ronaldtse** [comment]
I've rebased this branch.

**Comment by Intelligent2013** [comment]
@ronaldtse I've updated `annotate.yml` and fixed issues in `stepmod2mn` for

> - convert all XML files into adoc files for the path data/resource_docs/*/resource.xml
> - generate SVG images in the tree data

> - generate SVG image for Express Imagemap XML and image

This stepmod2mn feature intended for one SVG image generation for particular XML and image. The step `generate SVG images in the tree data` includes this one.

I don't understand what these commands do:

```
           # copy over per-schema EXPRESS-G files 
           git ls-files -- 'data/resources/*/*.svg' | xargs -I '{}' cp stepmod/{} {} 
  
           # copy over per-part EXPRESS-G files 
           for f in $(git ls-files -- 'data/documents/*/*.svg'); do \ 
             OLDPATH=${f//documents\/resources/resource_docs}; \ 
             cp stepmod/$OLDPATH $f; \ 
           done
```

The `git ls-files` doesn't return anything, because .svg didn't commit yet.
And what is the `stepmod/` folder?
What result would you like to achieve here?

**Comment by ronaldtse** [comment]
@Intelligent2013 I am trying to build a set of Metanorma documents that includes SVG files like this:
* https://github.com/metanorma/annotated-express/tree/main/data

In this repository, I only have 1 document (part 41, `fundamentals_of_product_description_and_support`):
* https://github.com/metanorma/annotated-express/tree/main/data/documents/resources/fundamentals_of_product_description_and_support

I want the annotation workflow to create all documents from the XML files.

**Comment by Intelligent2013** [comment]
@ronaldtse currently, 

- all 'resource.xml' convert into adoc, for instance:
`iso-10303-stepmod-wg12/data/resource_docs/fundamentals_of_product_description_and_support/resource.adoc`
`iso-10303-stepmod-wg12/data/resource_docs/fundamentals_of_product_description_and_support/sections/*.adoc`

- SVG images generated, for instance:
`iso-10303-stepmod-wg12/data/resource_docs/fundamentals_of_product_description_and_support/schema_diagexpg1.svg`

- all .svg copied into the folder `iso-10303-stepmod-wg12/stepmod`
(by the command `git ls-files --others 'data/resources/*/*.svg'  | xargs -I '{}' cp --parents {} stepmod`)

Is it enough?

I don't figure out what these commands do:
```
          # copy over per-part EXPRESS-G files
          for f in $(git ls-files -- 'data/documents/*/*.svg'); do \
            OLDPATH=${f//documents\/resources/resource_docs}; \
            cp stepmod/$OLDPATH $f; \
          done
```
In the repository `iso-10303-stepmod-wg12` the folder `documents` is missing.

**Comment by ronaldtse** [comment]
I think my remaining commands were just to live the SVG images to the right places. 

The next step would be to compile all the generated documents and make sure they work like Part 41. I guess we should test the flow on one part, let’s say, Part 41 because we can easily compare.

**Comment by Intelligent2013** [comment]
To test the generated documents, for begin, I've tried to test the generation for `annotated-express` document. And then copy the all environment .sh files generation into stepmod2mn. But currently, the metanorma generation failed for `annotated-express` (https://github.com/metanorma/annotated-express/issues/121). May be this issue relates to https://github.com/metanorma/annotated-express/issues/109 or https://github.com/metanorma/annotated-express/issues/97.
I'll try to investigate.

**Pull request closed without merging**